### PR TITLE
Adding benchmarking of various fingerprinting methods

### PR DESCRIPTION
**Pull request by ycombinator**
This PR adds benchmarking for various hash methods available in the fingerprint processor.

### Sample run

```
cd $GOPATH/src/github.com/elastic/beats/libbeat/processors/fingerprint
go test -bench=.
```

```
goos: darwin
goarch: amd64
pkg: github.com/elastic/beats/libbeat/processors/fingerprint
BenchmarkHashMethods/sha1-8             1000000000               0.0865 ns/op
BenchmarkHashMethods/sha256-8           1000000000               0.127 ns/op
BenchmarkHashMethods/sha384-8           1000000000               0.108 ns/op
BenchmarkHashMethods/sha512-8           1000000000               0.107 ns/op
BenchmarkHashMethods/xxhash-8           1000000000               0.0443 ns/op
BenchmarkHashMethods/md5-8              1000000000               0.0837 ns/op
PASS
ok      github.com/elastic/beats/libbeat/processors/fingerprint 7.093s
```

### CODE REVIEW

File: libbeat/processors/fingerprint/fingerprint_test.go [hunk 1]
```diff
@@ -376,3 +377,45 @@ func TestIgnoreMissing(t *testing.T) {
...
+	const charset = "abcdefghijklmnopqrstuvwxyz" +
+		"ABCDEFGHIJKLMNOPQRSTUVWXYZ" +
+		"0123456789"
+	charsetLen := len(charset)
+	b := make([]byte, 200)
+
+	var events []beat.Event
+	for i := 0; i < num; i++ {
+		for j := range b {
+			b[j] = charset[rand.Intn(charsetLen)]
```
Comment (urso) [inline libbeat/processors/fingerprint/fingerprint_test.go]: Benchmarks should be reproducible between runs, so to ensure they are comparable with future changes. Better create a PRNG using a constant seed, to ensure that the generated events are always the same: `rand.New(rand.NewSource(12345))`.

**Comment by urso** [comment]
Can you copy-paste a run as comment here?

**Comment by ycombinator** [comment]
Thanks @urso. I've addressed both points from your review feedback. The sample run output is in the PR description.

**Comment by urso** [comment]
Thanks. xxhash is quite fast, but then it is no crypto hash and seems to have a high chance of collision (only 64bit).

**Comment by ycombinator** [comment]
jenkins, test this

**Comment by ycombinator** [comment]
Travis CI is green and Jenkins CI failure looks unrelated. Merging.

**Pull request merged**